### PR TITLE
fix: Fix for issue 1120

### PR DESCRIPTION
**Pull request by systemmonkey42**
# Summary

Fixes `get_asdf_config_value_from_file()` to handle '=' in values
Fixes `run_hook()` to exit cleanly on hook failure.

Fixes: List issue numbers here
#1120

### CODE REVIEW

**Comment by jthegedus** [comment]
Can you add tests to this. 1 capturing the successful execution with the new code and 1 with an expected error with the usage that caused you to come and fix this?

**Comment by systemmonkey42** [comment]
> Can you add tests to this. 1 capturing the successful execution with the new code and 1 with an expected error with the usage that caused you to come and fix this?

More than happy to add the tests.

> It seems the usage outlined in #1120 is incorrect as the `asdfrc` file is only supposed to have values handled by the asdf-core, not the plugins.

Only asdf-core reads and executes the `pre_asdf_install_<plugin>` script.

The fact that it sets up the environment prior to the plugin installation process is arguably the intention of having a `pre_asdf_install_<plugin>` feature in the first place.  No plugin I'm aware of will read ~/.asdfrc directly.

**Comment by systemmonkey42** [comment]
I just found most tests 
```
325 tests, 216 failures
```
will fail if `ASDF_DATA_DIR` is set.
I might see if I can test in a docker container.

**Comment by systemmonkey42** [comment]
Test added.

File: lib/utils.bash [hunk 1]
```diff
@@ -658,7 +658,10 @@ asdf_run_hook() {
       unset asdf_hook_fun
       ev'al' "$hook_cmd" # ignore banned command just here
     }
-    asdf_hook_fun "${@:2}"
+    asdf_hook_fun "${@:2}" || {
+      display_error "Hook \"${hook_name}\" failed."
```
Comment (Stratus3D) [inline lib/utils.bash:662]: Is this consistent with other places in the codebase where hooks are invoked? I don't think we should print errors here but not in other places.

**Review by Stratus3D — APPROVED** [review]
Other than the one question I had, the code looks good to me. Build is running now.

File: lib/utils.bash (hunk 1 above)
Comment (systemmonkey42) [reply lib/utils.bash:662]: A quick survey of the code reveals that *all* hooks appear to end up in the `asdf_run_hook` function, and as such the error message and immediate exit _is_ consistent between all types of hooks.

However, I also recognize that a hook failure (like a post-reshim-plugin hook) which returns non-zero exit code might want to be ignored.

I am starting to feel that *not* modifying the asdf_run_hook, but rather checking the exit code on 
every call would be the better way to do it.

Eg, 
`lib/commands/command-exec.bash` already validates the exit code, and passes it on if the hook fails
```
      asdf_run_hook "pre_${plugin_name}_${shim_name}" "${shim_args[@]}"
      pre_status=$?
      if [ "$pre_status" -ne 0 ]; then
        return "$pre_status"
      fi
      exec "$executable_path" "${shim_args[@]}"
```
while the exit codes are ignored for 
`lib/commands/command-install.bash`
```
        asdf_run_hook "pre_asdf_download_${plugin_name}" "$full_version"
        "${plugin_path}"/bin/download
...
        asdf_run_hook "pre_asdf_install_${plugin_name}" "$full_version"
        "${plugin_path}"/bin/install
```

I suggest reverting the change to asdf_run_hook, and looking at checking the return code handling as a separate task

**Comment by systemmonkey42** [comment]
> Other than the one question I had, the code looks good to me. Build is running now.

Let me know how you feel about reverting the `asdf_run_hook` change.  The more I think about it, the more I suspect it will break existing behaviour

File: lib/utils.bash (hunk 1 above)
Comment (jthegedus) [reply lib/utils.bash:662]: >I suggest reverting the change to asdf_run_hook, and looking at checking the return code handling as a separate task
I concur

Comment (Stratus3D) [reply lib/utils.bash:662]: If you are saying change this:

```
 asdf_hook_fun "${@:2}" || {
      display_error "Hook \"${hook_name}\" failed."
```

back to this:

```
 asdf_hook_fun "${@:2}"
```

Then yes, I agree 👍

Comment (systemmonkey42) [reply lib/utils.bash]: Done

**Review by Stratus3D — APPROVED** [review]
@jthegedus any comments here? I'd like to get this merged.

**Comment by Stratus3D** [comment]
@systemmonkey42 I'm going to merge this PR shortly since it looks like @jthegedus isn't available right now. Can you update the PR to target the master branch instead of the `release-v0.10.0` branch? Thanks!

**Comment by Stratus3D** [comment]
I rebased and manually merged this branch into master. For some reason Github didn't detect the merge but I didn't modify your comments so you'll see them in the master branch now. Thanks for the PR!

**Pull request closed without merging**